src/App.js

```diff
@@ -15,30 +13,58 @@ function TodoList() {
   const [inputValue, setInputValue] = useState('');
 
   useEffect(() => {
+    const checkDueTimes = () => {
+      const now = new Date();
+      const updatedTodos = todos.map(todo => {
+        if (todo.dueTime!=null) {
+          const dueTime = new Date(todo.dueTime);
+          const timeDiff = dueTime - now;
+          if (timeDiff <= 0) {
+            todo.status = 'Overdue';
+          } else if (timeDiff <= 3600000) { //if more than 1hr
+            todo.status = 'DueWithIn1Hour';
+          } else if (timeDiff <= 7200000) { //if more than 2hr
+            todo.status = 'DueWithIn2Hour';
+          } else {
+            todo.status = 'Normal';
+          }
+        } else {
+          todo.status = 'NoDueDate';
```

Comment:
Eww, nested else ifs. Can you think of a better way to write this. Do not rewrite it. But can you think of a way to make it better?